tests/portals/github/test_manager.py

```diff
@@ -0,0 +1,404 @@
+# type: ignore
+import json
+import pytest
+from frictionless import portals, Catalog, Package, FrictionlessException
+
+
+# Read
+
+
+@pytest.mark.vcr
+def test_github_manager_read(options_without_dp):
+    url = options_without_dp.pop("url")
+    output = options_without_dp.pop("output")
+    package = Package(url)
+    assert json.loads(package.to_json()) == output
+
+
+@pytest.mark.vcr
+def test_github_manager_read_without_apikey(options_without_dp):
+    url = options_without_dp.pop("url")
+    output = options_without_dp.pop("output")
+    package = Package.from_github(url, control=portals.GithubControl(apikey=None))
+    assert json.loads(package.to_json()) == output
+
+
+@pytest.mark.vcr
+def test_github_manager_read_repo_with_datapackage(options_with_dp):
+    url = options_with_dp.pop("url")
+    package = Package(url)
+
+    # Read
+    expected_file_path = "data/datapackage.json"
+    with open(expected_file_path) as file:
+        assert json.loads(package.to_json()) == json.loads(file.read())
+
+
+@pytest.mark.vcr
+def test_github_manager_read_no_datapackage_found(options_empty):
+    url = options_empty.pop("url")
+    control = portals.GithubControl()
+    with pytest.raises(FrictionlessException) as excinfo:
+        Package.from_github(url, control=control)
+    error = excinfo.value.error
+    assert error.message == "Package/s not found"
+
+
+@pytest.mark.vcr
+def test_github_manager_read_only_csv_default(options_without_dp):
+    url = options_without_dp.pop("url")
+    output = options_without_dp.pop("output")
+    control = portals.GithubControl()
+    package = Package.from_github(url, control=control)
+    assert json.loads(package.to_json()) == output
+
+
+@pytest.mark.vcr
+def test_github_manager_read_other_file_types(options_without_dp):
+    url = options_without_dp.pop("url")
+    output = options_without_dp.pop("output_with_xls")
+    control = portals.GithubControl(formats=["csv", "xlsx"])
+    package = Package.from_github(url, control=control)
+    assert json.loads(package.to_json()) == output
+
+
+@pytest.mark.vcr
+def test_github_manager_read_with_url_and_control(options_without_dp):
+    url = options_without_dp.pop("url")
+    control = portals.GithubControl(formats=["xlsx"])
+    package = Package.from_github(url, control=control)
+    assert json.loads(package.to_json()) == {
+        "name": "test-repo-without-datapackage",
+        "resources": [
+            {
+                "name": "student",
+                "type": "table",
+                "path": "https://raw.githubusercontent.com/fdtester/test-repo-without-datapackage/master/data/student.xlsx",
+                "scheme": "https",
+                "format": "xlsx",
```

Comment:
It will be great if we can have `encoding` here also. Is it possible to get it from Github API metadata?